products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,114 @@
+
+### Debugging
+
+_How users can debug interactions with your system. When designing a system it's important to think about what tools you can provide to make debugging problems easier. Sometimes it's unclear whether the problem is in your system at all, so a mechanism for isolating a particular interaction and examining it to see if your system behaved as expected is very valuable. Once a system is in use, this is a great place to put tips and recipes for debugging. If this section grows too large, the mechanisms can be summarized here and individual tips can be moved to another document._
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+The search input would need to be sanitized.
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+Since the initial implementation will be an unauthenticated landing page, we will have little, if any, data about the user.
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII.
+
+### Open Questions and Risks
+
+- Should we index private documentation sources? If so, what private documentation sources can we expose to public users, if any?
+- Should the landing page be public only, or should it also support authenticated users?
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
+- What should the URL of the documentation landing page be? Since we have `design.va.gov`, does something like `docs.va.gov` make sense?
+- What is the current opportunity cost of building a custom solution? What are we choosing to delay right now by building a custom solution?
+- What is the future opportunity cost of maintaining a custom solution? What will we have to delay in the future by maintaining a custom solution?
+- What is the key person risk? How disruptive would it be if the author(s) of the custom solution were temporarily/permanently unavailable?
```

Comment:
These are really good questions! Building and maintaining a custom solution makes me nervous for a number of reasons. I don't know how to measure the opportunity costs myself, but my guess is it's not insignificant.

So we can have a point of reference, what do you anticipate the level of effort to be like for building and maintaining a custom solution? Like, what would be involved?

Reply:
My gut estimate is that it would take 1-3 sprints to build, deploy, and iterate on to get to a satisfactory state. But I suspect that could easily take much longer if we had to wait for approvals and/or other teams to set up the resources we're requesting.

Reply:
Do you have an idea for what maintenance on this would look like if we went with Algolia vs something else?

Reply:
It is worth clarifying in the design doc that we are recommending a custom _homepage_ solution — or, specifically for this case, a custom _search_ solution. ("Custom solution" is too vague IMO.)

Reply:
Also, is key person risk specific to this recommendation? I would think that would be more about our own docs about how to build and maintain any product, and less about whether we're recommending algolia or a different search solution?

Reply:
@meganhkelley Yeah, I was specifically asking about "Algolia vs [insert other fancy technology]" to accomplish the goal, not about the custom docs homepage as a whole. :+1:

What I'm trying to poke at is how much continual tweaking it'll take after it's all set up. If it's really fire-and-forget, awesome! I just tend to assume that's never the case. :sweat_smile:

Reply:
+++++ yep, I also assume that no-maintenance-forever is not the case, and this Q is definitely necessary to dig into!!

Reply:
It's difficult to speculate how much maintenance it would require, but my gut says that we'd probably want to budget 1-3 months worth of small adjustments, iterations, and fine tuning based on user feedback. I suspect after that, it would rarely need a lot of attention, unless a breaking change occurred (like renaming a repo, migrating a repo, or changes to a CSS selector, etc). 

Phrased another way, for the first few sprints after the MVP ships, perhaps we should budget a small percentage of the sprint capacity (<20%) for maintenance/iteration.